core/src/main/java/org/apache/spark/shuffle/sort/BypassMergeSortShuffleWriter.java

```diff
@@ -169,6 +173,36 @@ public void write(Iterator<Product2<K, V>> records) throws IOException {
       }
     }
     mapStatus = MapStatus$.MODULE$.apply(blockManager.shuffleServerId(), partitionLengths);
+    if (mapStatus instanceof HighlyCompressedMapStatus) {
+      HighlyCompressedMapStatus hc = (HighlyCompressedMapStatus) mapStatus;
+      long underestimatedBlocksSize = 0L;
+      for (int i = 0; i < partitionLengths.length; i++) {
+        if (partitionLengths[i] > mapStatus.getSizeForBlock(i)) {
+          underestimatedBlocksSize += partitionLengths[i];
+        }
+      }
+      writeMetrics.incUnderestimatedBlocksSize(underestimatedBlocksSize);
```

Comment:
This will essentially be sum of every block above average size - how is this supposed to be leveraged ?
For example:
1, 2, 3, 4, 5, 6 => 15
1, 2, 3, 4, 5, 10 => 15
(This ended up being a degenerate example - but in general, I am curious what the value is for this metric).